firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java

```diff
@@ -409,4 +424,23 @@ private String getErrorStream(HttpURLConnection connection) {
     }
     return errorStreamString;
   }
+
+  /** Gets the Android package's SHA-1 fingerprint. */
+  private String getFingerprintHashForPackage() {
+    byte[] hash;
+
+    try {
+      hash = AndroidUtilsLight.getPackageCertificateHashBytes(context, context.getPackageName());
+
+      if (hash == null) {
+        Log.e(TAG, "Could not get fingerprint hash for package: " + context.getPackageName());
+        return null;
+      } else {
+        return Hex.bytesToStringUppercase(hash, /* zeroTerminated= */ false);
+      }
+    } catch (PackageManager.NameNotFoundException e) {
+      Log.e(TAG, "No such package: " + context.getPackageName(), e);
+      return null;
```

Comment:
Should we be adding the header if the result is null?

Reply:
@vkryachko It's a good question, probably the backend responds the same way whether null or missing header. I was being consistent with the codebase here, which didn't have the null check.

https://github.com/firebase/firebase-android-sdk/blob/ac3276ce7d5f5181b1c419731495eb88a98537af/firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java#L246

Null, check would probably be more efficient on the wire, i.e not send a header if it doesn't exist. 

I can update it if you prefer the minor efficiency over the minor consistency reason. Let me know.

Reply:
I also realized since I am using the static function `AndroidUtilsLight.getPackageCertificateHashBytes(context, context.getPackageName())`, it would likely be easier to test with supporting null, and just check the presence of header.

as also done here.
https://github.com/firebase/firebase-android-sdk/blob/ac3276ce7d5f5181b1c419731495eb88a98537af/firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClientTest.java#L182

Reply:
@vkryachko I have addressed the comment in the update, thanks.